src/definition-finder.ts

```diff
@@ -0,0 +1,49 @@
+'use babel';
+
+import MouseMoveListener from './mouse-move-listener';
+import {SourceRange} from 'polymer-analyzer/lib/model/model'
+
+class DefinitionFinder extends MouseMoveListener {
+  sourcePosition: SourceRange;
+  removeMouseMoveListener: Function;
+
+  beforeUpdate() {
+    if (this.removeMouseMoveListener) {
+      this.removeMouseMoveListener();
+    }
+  };
+
+  obtainMouseMoveListener(textEditorElement: any) {
+    this.attachMouseDownListener(textEditorElement);
+    return async(event: MouseEvent) => {
+      const screenPosition = textEditorElement.component.screenPositionForMouseEvent(event);
+      const point = this.textEditor.bufferPositionForScreenPosition(screenPosition);
+      const relativePath: string =
+        atom.project.relativizePath(this.textEditor.getPath())[1];
+
+      try {
+        this.sourcePosition = await this.editorService.getDefinitionForFeatureAtPosition(relativePath, {line: point.row, column: point.column});
+      } catch (e) {
+        return;
+      }
+    };
+  };
+
+  private attachMouseDownListener(textEditorElement: any) {
+    console.log(textEditorElement);
+    const mouseDownListener = this.mouseDownListener.bind(this);
+    textEditorElement.addEventListener('mousedown', mouseDownListener);
+    this.removeMouseMoveListener = () => textEditorElement.removeEventListener('mousedown', mouseDownListener);
```

Comment:
Should this be a weakmap of textEditorElements to listeners? Otherwise we'll squash removeMouseMoveListener whenever we're called with a different one. Unless a new DefinitionFinder is created for each text editor, in which case shouldn't that be a property on the class?